womtool/src/main/scala/womtool/WomtoolMain.scala

```diff
@@ -139,6 +140,9 @@ object WomtoolMain extends App {
   val termination = runWomtool(args)
   termination.stdout foreach Console.out.println
   termination.stderr foreach Console.err.println
+
+  HttpResolver.sttpBackend.close()
```

Comment:
ToL: Sorry for turning a throwaway comment into a much bigger refactor, but if `sttpBackend` was not started naturally during processing then this line will have to create it (which will involve starting up an entire actor system) and then immediately shut it down again.

If that's not a negligible process, then this could be an annoying extra overhead on womtool processing time.

Reply:
That does sound quite silly.

It would be nice to somehow capture the startup / shutdown steps into `HttpResolver` itself so that its users don't have to remember (i.e. occasionally forget) to close the backend.

Is there a reasonable factoring for the above? I know destructors and the like are very much not a thing on the JVM

Reply:
This is a thing that looks like it could work: https://docs.oracle.com/javase/7/docs/api/java/lang/AutoCloseable.html

Reply:
I believe that that `AutoCloseable` is part of the try-with-resources syntax?

Reply:
Yeah, this would not be trivial

Reply:
I wonder if there's a nice scala way to check whether a `lazy val` has been instantiated yet to let us short-circuit the closing if it hasn't been instantiated yet

Reply:
Some interesting thoughts on https://stackoverflow.com/questions/17057255/how-to-check-if-a-lazy-val-is-initialized-without-initializing-it

Reply:
Was reading this too, love it:
>This isn't really an answer to your question, and I hate when people do this, but I'm going to do it anyway. I think the best response is: A lazy val isn't appropriate for this, so define a type that does support what you need.